specification/dashboard/resource-manager/Microsoft.Dashboard/stable/2022-08-01/grafana.json

```diff
@@ -0,0 +1,1091 @@
+          },
+          {
+            "$ref": "#/parameters/WorkspaceNameParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "in": "body",
+            "name": "requestBodyParameters",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/ManagedGrafana"
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK. Successfully create a workspace for Grafana resource.",
+            "schema": {
+              "$ref": "#/definitions/ManagedGrafana"
+            }
+          },
+          "201": {
+            "description": "Created. The operation was successfully started and will complete asynchronously.",
+            "schema": {
+              "$ref": "#/definitions/ManagedGrafana"
+            }
+          },
+          "default": {
+            "description": "Common error response for all Azure Resource Manager APIs to return error details for failed operations.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-long-running-operation": true,
+        "x-ms-long-running-operation-options": {
+          "final-state-via": "azure-async-operation"
+        },
+        "x-ms-examples": {
+          "Grafana_Create": {
+            "$ref": "./examples/Grafana_Create.json"
+          }
+        }
+      },
+      "patch": {
+        "tags": [
+          "GrafanaResource"
+        ],
+        "summary": "Update a workspace for Grafana resource.",
+        "operationId": "Grafana_Update",
+        "consumes": [
+          "application/json"
+        ],
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/WorkspaceNameParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "in": "body",
+            "name": "requestBodyParameters",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/ManagedGrafanaUpdateParameters"
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK. Successfully update the workspace for Grafana resource.",
+            "schema": {
+              "$ref": "#/definitions/ManagedGrafana"
+            }
+          },
+          "202": {
```

Comment:
>202 [](http://example.com/codeflow?start=11&length=3)

Is this status also returned to back-level clients? If so, this change needs breaking change review.

Reply:
Hi @mentat9 , thanks for checking this PR.
This is the initial PR for stable version API, do you mean it has breaking changes to old preview version? This 202 is the status code returned to ARM, since some newly added feature requires long running operation for update.

Reply:
Yes, breaking changes with respect to previous API version. Adding a new response status code breaks existing clients. The 202 is handled by the SDK code, but there won't be any code for it in your previous client. Ref: [https://microsoft.sharepoint.com/teams/azureecosystem/servicerings/Shared%20Documents/Breaking%20Changes/Azure%20Versioning%20and%20Breaking%20Changes%20-%20Policy%20Definition%20v1.2.3.pdf](https://microsoft.sharepoint.com/teams/azureecosystem/servicerings/Shared%20Documents/Breaking%20Changes/Azure%20Versioning%20and%20Breaking%20Changes%20-%20Policy%20Definition%20v1.2.3.pdf)

Reply:
Hi @mentat9 , thanks for the comment. There are two versions before,
1. `2021-09-01-preview`: There was no update supported. So, no breaking change.
2. ` 2022-05-01-preview`: We added update support, but this version is mainly for mock so that CLI team and we can test and validate the GA swagger spec to save time. Actually, although `2022-05-01-preview` is in public repo, we never officially released to any customer and our RP doesn't support it either. This is mainly for internal development purpose. So, we think it wouldn't introduce any breaking change in customer side.